File: class04/python/GQS0401-GeradorPlanTeste.py
#!/usr/bin/env python3
"""
GQS0401 - Gerador de Estrutura de Plano de Teste (IEEE 829).
Gera um template completo de plano de teste.
Uso: python3 GQS0401-GeradorPlanTeste.py
"""

TEMPLATE = """# Plano de Teste - {projeto}

**Versão:** 1.0 | **Data:** {data} | **Responsável:** {responsavel}

## 1. Identificador
PT-{projeto}-v1.0

## 2. Itens de Teste
{itens}

## 3. Funcionalidades a Testar
{funcionalidades_testar}

## 4. Funcionalidades a NÃO Testar
{funcionalidades_nao_testar}

## 5. Abordagem
- Níveis: {niveis}
- Técnicas: {tecnicas}
- Ferramentas: {ferramentas}

## 6. Critérios de Aprovação/Reprovação
{ criterios_aprovacao}

## 7. Critérios de Suspensão e Retomada
{suspensao}

## 8. Entregáveis
{entregaveis}

## 9. Recursos
{recursos}

## 10. Cronograma
{cronograma}

## 11. Riscos
{riscos}
"""


def gerar_plano():
    from datetime import date
    projeto = input("Nome do projeto: ") or "MeuProjeto"
    responsavel = input("Responsável: ") or "Eng. QA"
    data = date.today().isoformat()

    plano = TEMPLATE.format(
        projeto=projeto,
        data=data,
        responsavel=responsavel,
        itens="- Módulo de autenticação\n- Módulo de cadastro",
        funcionalidades_testar="- Login\n- Cadastro de usuário\n- Recuperação de senha",
        funcionalidades_nao_testar="- Relatórios (terceirizado)\n- Performance (fase separada)",
        niveis="Unitário, Integração, Sistema",
        tecnicas="Caixa-preta (partição de equivalência, BVA)",
        ferramentas="JUnit 5, pytest, Selenium",
        criterios_aprovacao="- 95% dos casos de teste passando\n- Nenhum bug crítico em aberto",
        suspensao="- Se ambiente de teste ficar indisponível por mais de 4h",
        entregaveis="- Plano de Teste\n- Casos de Teste\n- Relatório de Execução",
        recursos="- 1 Analista de Teste\n- 1 Engenheiro de Automação\n- Servidor Staging",
        cronograma="- Semana 1: Planejamento\n- Semana 2: Design\n- Semana 3-4: Execução",
        riscos="- Risco: indisponibilidade do ambiente\n- Mitigação: ambiente reserva em cloud",
    )
    print("\n" + "=" * 60)
    print(plano)
    print("=" * 60)


if __name__ == "__main__":
    print("GQS0401 - Gerador de Plano de Teste (IEEE 829)")
    gerar_plano()

File: class04/java/GQS0401-GeradorPlanTeste.java
/**
 * GQS0401 - Gerador de Estrutura de Plano de Teste (IEEE 829).
 * Gera um template completo de plano de teste.
 *
 * Compilação e execução:
 *   javac GQS0401-GeradorPlanTeste.java
 *   java -cp . GQS0401_GeradorPlanTeste
 */
import java.time.LocalDate;
import java.util.Scanner;

class GQS0401_GeradorPlanTeste {

    static final String TEMPLATE = """
        # Plano de Teste - %s

        **Versão:** 1.0 | **Data:** %s | **Responsável:** %s

        ## 1. Identificador
        PT-%s-v1.0

        ## 2. Itens de Teste
        - Módulo de autenticação
        - Módulo de cadastro

        ## 3. Funcionalidades a Testar
        - Login
        - Cadastro de usuário
        - Recuperação de senha

        ## 4. Funcionalidades a NÃO Testar
        - Relatórios (terceirizado)
        - Performance (fase separada)

        ## 5. Abordagem
        - Níveis: Unitário, Integração, Sistema
        - Técnicas: Caixa-preta (partição de equivalência, BVA)
        - Ferramentas: JUnit 5, pytest, Selenium

        ## 6. Critérios de Aprovação/Reprovação
        - 95%% dos casos de teste passando
        - Nenhum bug crítico em aberto

        ## 7. Critérios de Suspensão e Retomada
        - Se ambiente de teste ficar indisponível por mais de 4h

        ## 8. Entregáveis
        - Plano de Teste
        - Casos de Teste
        - Relatório de Execução

        ## 9. Recursos
        - 1 Analista de Teste
        - 1 Engenheiro de Automação
        - Servidor Staging

        ## 10. Cronograma
        - Semana 1: Planejamento
        - Semana 2: Design
        - Semana 3-4: Execução

        ## 11. Riscos
        - Risco: indisponibilidade do ambiente
        - Mitigação: ambiente reserva em cloud
        """;

    /** Lê os dados do projeto e imprime o plano de teste preenchido. */
    static void gerarPlano(Scanner sc) {
        System.out.print("Nome do projeto: ");
        String projeto = sc.nextLine().strip();
        if (projeto.isEmpty()) projeto = "MeuProjeto";

        System.out.print("Responsável: ");
        String responsavel = sc.nextLine().strip();
        if (responsavel.isEmpty()) responsavel = "Eng. QA";

        String data = LocalDate.now().toString();

        String plano = String.format(TEMPLATE, projeto, data, responsavel, projeto);

        System.out.println("\n" + "=".repeat(60));
        System.out.println(plano);
        System.out.println("=".repeat(60));
    }

    public static void main(String[] args) {
        System.out.println("GQS0401 - Gerador de Plano de Teste (IEEE 829)");
        try (Scanner sc = new Scanner(System.in)) {
            gerarPlano(sc);
        }
    }
}
